pkg/api/dictionaryinfo.go

```diff
@@ -0,0 +1,315 @@
+				DictionaryName: d.Name,
+				Digest:         info.Digest,
+				ItemCount:      float64(info.ItemCount),
+				LastUpdatedTS:  ts,
+			})
+		}
+	}
+	c.snapshot = out
+	level.Debug(c.logger).Log("msg", "refreshed dictionary info", "rows", len(out))
+	return nil
+}
+
+// dictionaryInfoCollector emits metrics from the DictionaryInfoCache snapshot.
+type dictionaryInfoCollector struct {
+	cache           *DictionaryInfoCache
+	itemCountDesc   *prometheus.Desc
+	lastUpdatedDesc *prometheus.Desc
+	digestInfoDesc  *prometheus.Desc
+}
+
+func (dc *dictionaryInfoCollector) Describe(ch chan<- *prometheus.Desc) {
+	ch <- dc.itemCountDesc
+	ch <- dc.lastUpdatedDesc
+	ch <- dc.digestInfoDesc
+}
+
+func (dc *dictionaryInfoCollector) Collect(ch chan<- prometheus.Metric) {
+	for _, r := range dc.cache.snapshot {
+		labels := []string{
+			r.ServiceID,
+			r.ServiceName,
+			strconv.Itoa(r.Version),
+			r.DictionaryID,
+			r.DictionaryName,
+		}
+		ch <- prometheus.MustNewConstMetric(dc.itemCountDesc, prometheus.GaugeValue, r.ItemCount, labels...)
+		ch <- prometheus.MustNewConstMetric(dc.lastUpdatedDesc, prometheus.GaugeValue, r.LastUpdatedTS, labels...)
+		digestLabels := append(labels, r.Digest)
+		ch <- prometheus.MustNewConstMetric(dc.digestInfoDesc, prometheus.GaugeValue, 1, digestLabels...)
+	}
+}
+
+// Gatherer returns a prometheus.Gatherer backed by a custom collector.
+func (c *DictionaryInfoCache) Gatherer(namespace, subsystem string) (prometheus.Gatherer, error) {
+	if !c.enabled {
+		return prometheus.Gatherers{}, nil
+	}
+
+	labelKeys := []string{"service_id", "service_name", "version", "dictionary_id", "dictionary_name"}
+
+	itemCountDesc := prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, subsystem, "dictionary_item_count"),
+		"Number of items in a Fastly edge dictionary.",
+		labelKeys, nil,
+	)
+
+	lastUpdatedDesc := prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, subsystem, "dictionary_last_updated_timestamp_seconds"),
+		"Unix timestamp (seconds) when the dictionary was last updated (UTC).",
+		labelKeys, nil,
+	)
+
+	digestInfoDesc := prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, subsystem, "dictionary_digest_info"),
+		"Digest of the dictionary content (info metric; value is always 1).",
+		append(labelKeys, "digest"), nil,
+	)
+
+	reg := prometheus.NewRegistry()
+	reg.MustRegister(&dictionaryInfoCollector{
+		cache:           c,
+		itemCountDesc:   itemCountDesc,
+		lastUpdatedDesc: lastUpdatedDesc,
+		digestInfoDesc:  digestInfoDesc,
+	})
+
+	return reg, nil
+}
+
+// --- Fastly API helpers ---
+
+func (c *DictionaryInfoCache) req(ctx context.Context, method, url string) (*http.Request, error) {
+	req, err := http.NewRequestWithContext(ctx, method, url, nil)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("Fastly-Key", c.token)
+	req.Header.Set("Accept", "application/json")
+	return req, nil
+}
+
+func (c *DictionaryInfoCache) listServices(ctx context.Context) ([]svc, error) {
```

Comment:
I'm not super familiar with this codebase, but there's an existing `serviceCache` that's already refreshing the service list. Is there a reason that we can't use that in this file? That would avoid having two different caches refreshing the service list. I think using that would also respect the `service`, `service-allowlist`, `service-blocklist`, and `service-shard` flags.